Scripts/extras/package_downloader.py

```diff
@@ -0,0 +1,134 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+# 
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import os
+import urllib
+import urllib.request
+import ssl
+import certifi
+import hashlib
+import pathlib
+import tarfile
+import sys
+
+class PackageDownloader(): 
+    @staticmethod
+    def DownloadAndUnpackPackage(package_name, package_hash, folder_target):
+        ''' given a public server URL list (semicolon-seperated), 
+            finds the package, if possible, and downloads and unpacks it to a given folder.
+            Note that this essentially mimics the server urls protocol used by cmake on the client
+            side and thus will also check buckets if s3:// urls are given
+            and will stop at the first one it finds.
+
+            Assumes that LY_PACKAGE_SERVER_URLS is set in the environment
+            Assumes that, if necessary, LY_AWS_PROFILE is set in the environment (if using s3 buckets)
+
+            returns True if it succeeded, False otherwise.
+         '''
```

Comment:
```suggestion
        '''Given a semicolon-separated public server URL list (semicolon-seperated), 
            attempts to find the package. If found, downloads and unpacks to the target_folder location.
            Only the first found package is downloaded, and then the search stops. If the checksum of
            the downloaded file doesn't match the checksum in the O3DE dependency list, the package
            isn't unpacked on the filesystem and the download is deleted.
        
            This method supports all URI types handled by the O3DE package system, including S3 URIs.
            
            PRECONDITIONS:
            * LY_PACKAGE_SERVER_URLS must be set in the environment
            * If using S3 URIs, LY_AWS_PROFILE must be set in the environment
            
            Returns True if successful, False otherwise.
         '''
```